### DISCUSSION TITLE
[v4] Using `next/font` variables in the theme config

### DISCUSSION THREAD

**Discussion by mikeploythai** [tailwindlabs/tailwindcss#15267] [Help] [answered]
I'm trying to utilize my fonts from `next/font/google` via their CSS variables in the `@theme` config to override `--font-sans`, and to create a new variable `--font-heading`

```css
// globals.css
@import 'tailwindcss';

@theme {
  --font-heading: var(--font-unbounded);
  --font-sans: var(--font-montserrat);
}
```

```tsx
// layout.tsx
const unbounded = Unbounded({
  subsets: ['latin'],
  variable: '--font-unbounded',
})

const montserrat = Montserrat({
  subsets: ['latin'],
  variable: '--font-montserrat',
})

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode
}>) {
  return (
    <html lang="en">
      <body className={`${unbounded.variable} ${montserrat.variable} antialiased`}>{children}</body>
    </html>
  )
}
```

```tsx
// page.tsx
export default function Home() {
  return (
    <main>
      <h1 className="font-heading">Test heading</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum,
        provident.
      </p>
    </main>
  );
}
```

~~To my understanding, the code above doesn't work because of how Tailwind doesn't know of any external CSS variables other than its own or anything custom that's written directly in the CSS file ([ref](https://github.com/tailwindlabs/tailwindcss/discussions/15113)).~~

EDIT: re-read the discussion post and found out that was more of a media query issue, heh

<img width="328" alt="image" src="https://github.com/user-attachments/assets/07e5cb29-5d47-4953-bf43-d4db9cf3f207">

<img width="328" alt="image" src="https://github.com/user-attachments/assets/36cad2a1-17ea-40c5-8809-27c761870d1d">

> Example of the variables not defined

In this case, how can I go about using fonts from `next/font` in my theme config? There was a [similar question](https://github.com/tailwindlabs/tailwindcss/discussions/13410) from an early alpha that seems to address this, but I can't get it to work the way I expect it to.

**Comment by wongjn** [comment]
Try altering your CSS like:

```diff

-@theme {
+@theme inline {
   --font-heading: var(--font-unbounded);
   --font-sans: var(--font-montserrat);
```

**Comment by adamwathan** [comment, marked as answer]
Instead of setting the variable classes on the body tag, put them on the html tag 👍🏻

**Reply by mikeploythai** [reply]
thank you adam 🙏
